packages/@aws-cdk/cloudformation-diff/lib/diff-template.ts

```diff
@@ -93,7 +104,11 @@ function propagatePropertyReplacement(source: types.ResourceDifference, dest: ty
   }
 }
 
-function calculateTemplateDiff(currentTemplate: { [key: string]: any }, newTemplate: { [key: string]: any }): types.TemplateDiff {
+function calculateTemplateDiff(
+  currentTemplate: { [key: string]: any },
+  newTemplate: { [key: string]: any },
+  replacements?: ResourceReplacements,
```

Comment:
I'm wondering if we can avoid passing this structure all the way down through a bunch of functions.

For example:

- Either stick all of these on a class, so it becomes a class member.
- Or: organize the diffing more into a pipeline. First we calculate all the diffs that derive from the template change itself, and then using the information from change set we remove false positives from the diff. So it would be like:

```ts
function fullDiff(a, b, changeSet) {
  return filterFalsePositivies(changeSet, templateDiff(a,b));
}
```

It might be that that is not possible (tell me if it isn't). But I like keeping concerns separated as much as possible.

Reply:
The second approach was much easier than my first approach 😅